src/problem.cpp

```diff
@@ -181,9 +181,10 @@ Problem::FindSolutions(Handle& handle, const FindOptions& options, std::size_t m
             [&](const ConvolutionDescriptor& op_desc) {
                 if(op_desc.mode == miopenTranspose)
                     return MakeTransposed().FindSolutionsImpl(
-                        handle, options, max_solutions, buffers, op_desc);
+                        handle, options, max_solutions, buffers, op_desc, *this);
```

Comment:
I can see why the original change started causing issues since we would be giving back the transposed problem with the solution, and that can cause issues later on.

However, for the reverted version from #3388 why was it causing issues / fixed others?
These methods are const, and it seems like FindSolutionsImpl must be changing state despite being const.

Reply:
MakeTransposed() produces a pre-transposed miopen::Problem, which has different direction and swapped descriptors compared to *this

IIRC, reverted version actually has several problems: ValidateGroupCount is called with y descriptor instead of x and something is swapped incorrectly, but I do not remember what exactly

Current approach combines actually correct behavior of FindSolutionImpl from 3285 (the bug there was storing wrong problem) with original problem stored in the solution, thus RunSolution also having correct behavior.

Reply:
From the revert PR it looks like the swap was happening, but I could believe that it wasn't working as expected, which lead to knock on issues which this version will resolve.